ipaserver/install/ipa_backup.py

```diff
@@ -357,10 +356,11 @@ def run(self):
                                  options.gpg_keyring)
 
         finally:
+            tmp_dir = tempfile.gettempdir()
             try:
-                os.chdir(cwd)
+                os.chdir(tmp_dir)
             except Exception as e:
-                logger.error('Cannot change directory to %s: %s', cwd, e)
+                logger.error('Cannot change directory to %s: %s', tmp_dir, e)
```

Comment:
I'm not sure another temporary directory is needed here, or perhaps any chdir at all. The previous chdir was needed because it did a chdir(self.dir) so in order to clean up one needed to chdir out. Since chdir(self.dir) isn't done any more there is no need to change to somewhere else.

Reply:
You might be right. I left it in since I didn't play enough close attention what self.top_dir actually is. I thought it might be possible to start the script from inside it and then the `chdir` would be necessary. Since it is a `tempfile.mkdtemp` this definitely isn't possible. I'll test the change and push an update.

Reply:
Removed and tested. Seems to work just fine.